GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -0,0 +1,246 @@
+        /// <returns>The path extracted from the provided line of the git index</returns>
+        private string TrimGitIndexLine(string line)
+        {
+            return line.Substring(line.IndexOf('\t') + 1);
+        }
+
+        /// <summary>
+        /// Talk to the mount process across the named pipe to get a list of the modified paths
+        /// </summary>
+        /// <param name="enlistment">The enlistment being operated on</param>
+        /// <returns>An array containing all of the modified paths in string format</returns>
+        private string[] GetModifiedPathsFromPipe(GVFSEnlistment enlistment)
+        {
+            using (NamedPipeClient pipeClient = new NamedPipeClient(enlistment.NamedPipeName))
+            {
+                string[] modifiedPathsList = { };
+
+                if (!pipeClient.Connect())
+                {
+                    this.ReportErrorAndExit("Unable to connect to GVFS.  Try running 'gvfs mount'");
+                }
+
+                try
+                {
+                    NamedPipeMessages.Message modifiedPathsMessage = new NamedPipeMessages.Message(NamedPipeMessages.ModifiedPaths.ListRequest, NamedPipeMessages.ModifiedPaths.CurrentVersion);
+                    pipeClient.SendRequest(modifiedPathsMessage);
+
+                    NamedPipeMessages.Message modifiedPathsResponse = pipeClient.ReadResponse();
+                    if (!modifiedPathsResponse.Header.Equals(NamedPipeMessages.ModifiedPaths.SuccessResult))
+                    {
+                        this.Output.WriteLine("Bad response from modified path pipe: " + modifiedPathsResponse.Header);
+                        return modifiedPathsList;
+                    }
+
+                    modifiedPathsList = modifiedPathsResponse.Body.Split(new char[] { '\0' }, StringSplitOptions.RemoveEmptyEntries);
+                }
+                catch (BrokenPipeException e)
+                {
+                    this.ReportErrorAndExit("Unable to communicate with GVFS: " + e.ToString());
+                }
+
+                return modifiedPathsList;
+            }
+        }
+
+        /// <summary>
+        /// Get two lists of placeholders, one containing the files and the other the directories
+        /// Goes to the SQLite database for the placeholder lists
+        /// </summary>
+        /// <param name="enlistment">The current GVFS enlistment being operated on</param>
+        /// <param name="filePlaceholders">Out parameter where the list of file placeholders will end up</param>
+        /// <param name="folderPlaceholders">Out parameter where the list of folder placeholders will end up</param>
+        private void GetPlaceholdersFromDatabase(GVFSEnlistment enlistment, out List<IPlaceholderData> filePlaceholders, out List<IPlaceholderData> folderPlaceholders)
+        {
+            GVFSDatabase database = new GVFSDatabase(new PhysicalFileSystem(), enlistment.EnlistmentRoot, new SqliteDatabase());

```

Comment:
You need a `using()` block here.

Reply:
An aside: I saw a very small warning underline for the `new` here, and maybe it is triggering [rule CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope?view=vs-2019). I wonder if we could enforce this during our build?